### PR TITLE
Adding Github private repo test

### PR DESCRIPTION
**Pull request by mmartin24**
Automating part 3 of https://github.com/rancher/fleet-e2e/issues/51

### CODE REVIEW

File: .github/workflows/ui-rm_head_2.7.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ jobs:
       gitlab_private_pwd: ${{ secrets.GITLAB_PRIVATE_PWD }}
       bitbucket_private_user: ${{ secrets.BITBUCKET_PRIVATE_USER }}
       bitbucket_private_pwd: ${{ secrets.BITBUCKET_PRIVATE_PWD }}
+      gh_private_user: ${{ secrets.GH_PRIVATE_USER }}
+      gH_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```
Comment (thehejik) [inline .github/workflows/ui-rm_head_2.7.yaml:40]:
```suggestion
      gh_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```

File: .github/workflows/ui-rm_head_2.8.yaml [hunk 2]
```diff
@@ -36,6 +36,8 @@ jobs:
       gitlab_private_pwd: ${{ secrets.GITLAB_PRIVATE_PWD }}
       bitbucket_private_user: ${{ secrets.BITBUCKET_PRIVATE_USER }}
       bitbucket_private_pwd: ${{ secrets.BITBUCKET_PRIVATE_PWD }}
+      gh_private_user: ${{ secrets.GH_PRIVATE_USER }}
+      gH_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```
Comment (thehejik) [inline .github/workflows/ui-rm_head_2.8.yaml:40]:
```suggestion
      gh_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```

File: .github/workflows/ui-rm_head_2.9.yaml [hunk 3]
```diff
@@ -36,6 +36,8 @@ jobs:
       gitlab_private_pwd: ${{ secrets.GITLAB_PRIVATE_PWD }}
       bitbucket_private_user: ${{ secrets.BITBUCKET_PRIVATE_USER }}
       bitbucket_private_pwd: ${{ secrets.BITBUCKET_PRIVATE_PWD }}
+      gh_private_user: ${{ secrets.GH_PRIVATE_USER }}
+      gH_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```
Comment (thehejik) [inline .github/workflows/ui-rm_head_2.9.yaml:40]:
```suggestion
      gh_private_pwd: ${{ secrets.GH_PRIVATE_PWD }}
```

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts [hunk 4]
```diff
@@ -164,5 +164,42 @@ describe('Fleet Deployment Test Cases', () => {
...
+    it('FLEET-8: Test GITHUB Private Repository to install NGINX app using HTTP auth', () => {
+      const repoName = "local-cluster-fleet-8"
+      const branch = "main"
+      const path = "nginx"
+      const repoUrl = "https://github.com/mmartinsuse/test-fleet"
+      const gitAuthType = "http"
+      const userOrPublicKey = Cypress.env("gh_private_user");
+      const pwdOrPrivateKey = Cypress.env("gh_private_pwd");
+  
+      // // Click on the Continuous Delivery's icon
```
Comment (thehejik) [inline tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:177]:
```suggestion
      // Click on the Continuous Delivery's icon
```

Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:177]: thanks; noted

**Review by thehejik — APPROVED** [review]
LGTM

**Pull request merged**